chapter9.adoc

```diff
@@ -0,0 +1,334 @@
+
+The `RCID` and `MCID` accompany each request made by the hart, with effective
+privilege mode equal to M, to shared resource controllers . The `RCID` is used to
+determine the resource allocations (e.g., cache occupancy limits, memory
+bandwidth limits, etc.) to enforce. The `MCID` is used to identify a counter to
+monitor resource usage.
+
+[NOTE]
+====
+A suggested reset value of 0 for the `RCID` field within the `mrmcfg` register
+aligns with the default operational behavior of resource controllers, which is
+to associate all capacity and resources with `RCID=0`. This setup ensures a
+seamless initial state that conforms to expected default resource allocation
+policies. The reset value for the `MCID` field, being potentially
+implementation-defined, does not directly impact core functionality and may
+vary based on specific system requirements.
+====
+
+When Smqosid is implemented, the configurations in the `srmcfg` register
+introduced by Ssqosid extension only apply to requests made with effective
+privilege modes less than M.
+
+[[MRMCFG64]]
+.Machine Resource Management Configuration (`mrmcfg`) register for MXLEN=64
+
+[wavedrom, , ]
+....
+{reg: [
+  {bits: 12, name: 'RCID'},
+  {bits:  4, name: 'WPRI'},
+  {bits: 12, name: 'MCID'},
+  {bits: 36, name: 'WPRI'},
+], config:{lanes: 2, hspace:1024}}
+....
+
+[[MRMCFG32]]
+.Machine Resource Management Configuration (`mrmcfg`) register for MXLEN=32
+
+[wavedrom, , ]
+....
+{reg: [
+  {bits: 12, name: 'RCID'},
+  {bits:  4, name: 'WPRI'},
+  {bits: 12, name: 'MCID'},
+  {bits:  4, name: 'WPRI'},
+], config:{lanes: 1, hspace:1024}}
+....
+
+The Smqosid extension introduces a supervisor-RCID-length (`SRL`) and a
+supervisor-MCID-length (`SML`) field in the `msdcfg` register. The `SRL` and
+`SML` fields are used to configure a range of `RCID` and `MCID` that may be used
+by a supervisor domain. The `SRL` and `SML` fields are WARL and their legal
+values range from 0 to 12. The Smqosid extension additionally introduces a
+supervisor-srmcfg-mode (`SSM`) field. The legal values of this mode field are 0
+and 1, which,  along with `SRL` and `SML` determine the range of `RCID` and
+`MCID` values that the SD is allowed to program into the `srmcfg` register.
+
+.`msdcfg` register
+
+[wavedrom, , ]
+....
+{reg: [
+  {bits:  6, name: 'SDICN'},
+  {bits:  1, name: 'SSM'},
+  {bits:  1, name: 'SDEDBGALW'},
+  {bits:  4, name: 'SRL'},
+  {bits:  4, name: 'SML'},
+  {bits:  4, name: 'SQRID'},
+  {bits:  4, name: 'MQRID'},
+  {bits:  8, name: 'WPRI'},
+], config:{lanes: 4, hspace:1024}}
+....
+
+When the `srmcfg` CSR is read at privilege modes less than M, the value returned
+for the `RCID` and `MCID` fields of the register is computed using <<eq-1>>.
+
+[latexmath#eq-1,reftext="equation ({counter:eqs0})"]
+++++
+\begin{equation}
+RCID-value = (SSM == 1) ? (srmcfg.RCID & ((1 << SRL) - 1)) : srmcfg.RCID
+MCID-value = (SSM == 1) ? (srmcfg.MCID & ((1 << SML) - 1)) : srmcfg.MCID
+\end{equation}
```

Comment:
please split these into separate equations else they render on the same line (in very very tiny font)

Reply:
Fixed